src/os/bluestore/BlueStore.cc

```diff
@@ -11826,4 +12230,217 @@ void BlueStore::_apply_padding(uint64_t head_pad,
   }
 }
 
+void BlueStore::_record_onode( OnodeRef o, KeyValueDB::Transaction txn)
```

Comment:
`KeyValueDB::Transaction` is `ceph::shared_ptr<TransactionImpl>`. I would love to see it someday renamed to e.g. `TransactionRef`. Currently the naming clashes with `ObjectStore::Transaction` and suggests taking by value.

Maybe just take `o` and `txn` by reference? The method doesn't seem to have any special requirements on params' lifecycle. As a tiny bonus we wouldn't need to update ref counters.